src/harness/sas_test_harness.py

```diff
@@ -284,71 +284,74 @@ def generateCbsdReferenceId(fcc_id, serial_number):
+    encoded_cbsd_id = generateCbsdReferenceId(registration_request['fccId'],
+                                         registration_request['cbsdSerialNumber'])
 
-    # Calculate grant_expiry time for grant objects if grant expiry time may
-    # be incorrect or missing.
+    # Assign grant expiry time to grant objects.
     grant_time_stamp = datetime.now().today() + timedelta(hours=24)
-    grant_expiry_time = time.strftime(grant_time_stamp.replace(microsecond=0).isoformat()) + 'Z'
+    grant_expire_time = time.strftime(grant_time_stamp.replace(microsecond=0).isoformat()) + 'Z'
 
     internal_data = {
         'id': 'cbsd/{}'.format(encoded_cbsd_id),
         'registration': {
-            'fccId': cbsd_record['fccId'],
-            'cbsdCategory': cbsd_record['cbsdCategory'],
-            'callSign': cbsd_record['callSign'],
-            'userId': cbsd_record['userId'],
-            'airInterface': cbsd_record['airInterface'],
-            'measCapability': cbsd_record['measCapability'],
-            'installationParam': cbsd_record['installationParam']
+            'fccId': registration_request['fccId'],
+            'cbsdCategory': registration_request['cbsdCategory'],
+            'callSign': registration_request['callSign'],
+            'airInterface': registration_request['airInterface'],
+            'measCapability': registration_request['measCapability'],
+            'installationParam': registration_request['installationParam']
         }
     }
 
     # Check if groupingParam optional parameter present in CBSDRecord then
     # add to registration object.
-    if cbsd_record.has_key('groupingParam'):
-      internal_data['registration'].update({'groupingParam': cbsd_record['groupingParam']})
-    internal_data_grant = {}
-    internal_data_grant['grants'] = []
+    if registration_request.has_key('groupingParam'):
+      internal_data['registration'].update({'groupingParam': registration_request['groupingParam']})
+
+    internal_data['grants'] = []
 
     # Read all grant records if mulitiple grant records are associated with single cbsd.
-    for grant_record in grant_records:
+    for grant_request in grant_requests:
       grant_data = {}
-      if not grant_record.has_key('id'):
-        grant_data['id'] = 'SAMPLE_ID_{:05}'.format(random.randrange(1, 10 ** 5))
-      else:
-        grant_data['id'] = grant_record['id']
-      grant_data['operationParam'] = grant_record['operationParam']
-      if not grant_record.has_key('channelType'):
-        grant_data['channelType'] = 'GAA'
-      else:
-        grant_data['channelType'] = grant_record['channelType']
-      if not grant_record.has_key('grantExpireTime'):
-        grant_data['grantExpireTime'] = grant_expiry_time
+      if grant_request.has_key('id'):
+        grant_data['id'] = grant_request['id']
+      if grant_request.has_key('operationParam'):
+        grant_data['operationParam'] = grant_request['operationParam']
```

Comment:
operationParam is not an optional field in SAS-SAS exchange. You should probably assert (or assume) that the grant request has this information, otherwise you may accidentally give invalid information to the SAS UUT.

Reply:
Added assertion if operationParam is not in grant request